src/ray/gcs/gcs_server/gcs_heartbeat_manager.h

```diff
@@ -89,7 +94,10 @@ class GcsHeartbeatManager : public rpc::HeartbeatInfoHandler {
   /// The callback of node death.
   std::function<void(const NodeID &)> on_node_death_callback_;
   /// The number of heartbeats that can be missed before a node is removed.
-  int64_t num_heartbeats_timeout_;
+  const int64_t num_heartbeats_timeout_;
+  /// The initial heartbeat timeout when GCS restarts and waiting for raylet to
+  /// reconnect. Once connected, we'll use num_heartbeats_timeout_
+  const int64_t initial_num_heartbeats_timeout_;
```

Comment:
the name is a bit confusing, how about?  `gcs_failover_worker_reconnect_timeout_` ?